gateway-provider-rewrite-common/src/main/java/org/apache/knox/gateway/filter/rewrite/impl/xml/XmlUrlRewriteRulesExporter.java

```diff
@@ -59,7 +59,11 @@ public String getFormat() {
   }
 
   @Override
-  public void store( UrlRewriteRulesDescriptor descriptor, Writer writer ) throws IOException {
+  public Object store(UrlRewriteRulesDescriptor descriptor, Writer writer) throws IOException {
+    return store(descriptor, writer, false);
+  }
+
+  public Object store( UrlRewriteRulesDescriptor descriptor, Writer writer, boolean omitXmlHeader ) throws IOException {
```

Comment:
Can't this be Document?

Reply:
No, JAXB does not handle interfaces.

Reply:
No, because of this class implements `UrlRewriteRulesExporter`. As of now, only one implementation exists, but the general contract - defined by the interface - that you store something and return it (e.g. Document in case of XML, String in case of JSON).